src/read_write/codec.rs

```diff
@@ -39,25 +40,25 @@ impl PositionEncoding {
     }
 
     // TODO(sirver): Returning a Result here makes this function more expensive than needed - since
-    // we require stack space for the full Result. This shuold be fixable to moving to failure.
-    pub fn from_proto(proto: proto::Node_PositionEncoding) -> Result<Self> {
+    // we require stack space for the full Result. This should be fixable to moving to failure.
+    pub fn from_proto(proto: proto::PositionEncoding) -> Result<Self> {
         match proto {
-            proto::Node_PositionEncoding::INVALID => {
-                Err(ErrorKind::InvalidInput("Invalid PositionEncoding".to_string()).into())
+            proto::PositionEncoding::Uint8 => Ok(PositionEncoding::Uint8),
+            proto::PositionEncoding::Uint16 => Ok(PositionEncoding::Uint16),
+            proto::PositionEncoding::Float32 => Ok(PositionEncoding::Float32),
+            proto::PositionEncoding::Float64 => Ok(PositionEncoding::Float64),
+            proto::PositionEncoding::INVALID => {
+                Err(ErrorKind::Proto("Proto: PositionEncoding is invalid".to_string()).into())
```

Comment:
Do we really need a new error type for this or can we just leave it with `InvalidInput`?

Reply:
the description of invalid input says ("The input file is not supported or invalid") does not fit really good. shall I remove "file" from the description?

Reply:
Yep, let's just remove file and stick with the old error type.